### ISSUE TITLE
Regression 0.5.0 -> 0.6.0 - title doesn't display as expected

### ISSUE THREAD

**Issue by vsajip** [Textualize/textual#1366]
The following script
```python
from __future__ import annotations

from rich.text import Text
from textual.app import App, ComposeResult
from textual.containers import Container, Horizontal
from textual.reactive import Reactive, watch
from textual.widget import Widget

class TitleWidget(Widget):
    """Display a title."""

    DEFAULT_CSS = '''
    TitleWidget {
        content-align: center middle;
        width: 100%;
        background: $secondary-background;
        height: 1;
    }
    '''

    text: Reactive[str] = Reactive('')

    def render(self) -> Text:
        return Text(self.text, no_wrap=True, overflow='ellipsis')

class AppHeader(Widget):
    """
    A header widget for the application.
    """

    DEFAULT_CSS = '''
    AppHeader {
        dock: top;
        width: 100%;
        background: $secondary-background;
        color: $text;
        height: 1;
    }
    '''

    DEFAULT_CLASSES = ''

    def __init__(
        self,
        *,
        name: str | None = None,
        id: str | None = None,
        classes: str | None = None,
    ):
        super().__init__(name=name, id=id, classes=classes)
        self.title = TitleWidget()

    def compose(self):
        yield self.title

    def on_mount(self) -> None:

        def set_title(title: str) -> None:
            self.title.text = title

        watch(self.app, 'title', set_title)

class TestApp(App):

    CSS = '''
        AppHeader {
            border-bottom: black;
        }
    '''

    def __init__(self):
        super().__init__()
        self.title = 'Test App Title'

    def compose(self) -> ComposeResult:
        yield AppHeader()
        yield Horizontal(Container(), Container())

app = TestApp()
app.run()
```

used to work in 0.5.0, displaying

![ss20](https://user-images.githubusercontent.com/130553/207672000-a4107f3e-9cff-4f68-9db2-5316923cc86f.png)

However,  in 0.6.0, the title isn't displayed, and nor is the border:

![ss21](https://user-images.githubusercontent.com/130553/207672243-82326b73-f436-4f1d-8bc3-413a02984d07.png)

The problem appears to be connected to the ` border-bottom: black;` CSS. If that's commented out, then on both 0.5.0 and 0.6.0 you get

![ss22](https://user-images.githubusercontent.com/130553/207672786-ea30d3ef-ff88-4ca5-8b61-a09b6ea6a674.png)

This is on Linux Mint 64-bit, Cinnamon desktop.

**Comment by davep** [comment]
As best as I can tell, the issue you have here is you're setting your header widget to be `height: 1`, but then also putting a border on it too. This would mean that the space available for your title, the padding and the border just isn't there (given that [the default is that the padding and the border are included in the height calculation](https://textual.textualize.io/styles/box_sizing/#box-sizing).

If I add `box-sizing: content-box;` to the CSS for your `AppHeader` it works as you would expect.

As for the difference between 0.5.0 and 0.6.0... I can't say what the exact difference is off the top of my head, but 0.6.0 [did include some fixes and improvements to sizing](https://github.com/Textualize/textual/blob/main/CHANGELOG.md#060---2022-12-11) which may have fixed an incorrect calculation that existed before; possibly.

In summary: if I'm reading the docs correctly then it wasn't working correctly in 0.5.0 but is now.

**Comment by davep** [comment]
Just to compare things between 0.5.0 and 0.6.0, forgive the rather ugly code, but it's a quick test hack:

```python
from textual.app import App, ComposeResult
from textual.containers import Horizontal
from textual.widgets import Static

class TestApp( App[ None ] ):

    CSS = "Horizontal { height: 1fr;}\n.border-calc {box-sizing: content-box; }\nStatic { border: solid red}\n" + "\n".join(
        f"Static.box-{n} {{ width: {(1+n)*2}; height: {1+n};}}"
        for n in range( 10 )
    )

    def compose( self ) -> ComposeResult:
        yield Horizontal(
            *[ Static( str(n), classes=f"box-{n}" ) for n in range( 10 ) ]
        )
        yield Horizontal(
            *[ Static( str(n), classes=f"border-calc box-{n}" ) for n in range( 10 ) ]
        )

if __name__ == "__main__":
    TestApp().run()
```

With 0.6.0 I get this:

![Screenshot 2022-12-15 at 20 51 36](https://user-images.githubusercontent.com/28237/207964748-40a81dca-5006-42ae-a62f-9ce40293c338.png)

and with 0.5.0 this:

![Screenshot 2022-12-15 at 20 53 21](https://user-images.githubusercontent.com/28237/207964980-eefecf2b-f8c3-4782-9594-ac76e1d994bb.png)

0.6.0 looks to better match what's documented, I think.

**Comment by vsajip** [comment]
Thanks for the explanation and illustrative examples.
